apps/condo/domains/property/components/UnitButton.tsx

```diff
@@ -0,0 +1,159 @@
+        background-color: #F5F5F5;
+        color: black;
+        border-color:  black;
+    }
+    &:focus {
+        background-color: #F5F5F5;
+        color: black;
+        border-color:  #F5F5F5;
+    }
+
+    &:active {
+        background-color: #F5F5F5;
+        color: black;
+        border-color: black;
+    }
+
+    &:disabled, &:disabled:hover {
+        background-color: #F5F5F5;
+        cursor: default;
+        color: black;
+        border: 1px solid #F5F5F5;
+    }
+ `
+
+const buttonSecondaryCss = css`
+    display: inline-block;
+    background-color: transparent;
+    border: 1px solid transparent;
+    color: black;
+    font-size: 12px;
+    line-height: 40px;
+    border-radius: 5px;
+    margin-top: 2px;
+    margin-right: 2px;
+    text-align: center;
+    height: 40px;
+    width: 40px;
+    padding: 0px;
+    box-shadow: none;
+
+    &:hover {
+        background-color: transparent;
+        color: black;
+        border-color: black;
+    }
+    &:focus {
+        background-color: transparent;
+        color: black;
+        border-color: transparent;
+    }
+
+    &:active {
+        background-color: transparent;
+        color: black;
+        border-color: black;
+    }
+
+    &:disabled, &:hover:disabled {
+        background-color: transparent;
+        color: black;
+        border-color: transparent;        
+        cursor: default;
+    }    
+`
+const selectedCss = css`
+    background-color: black;
+    color: white;
+    border-color: black;
+    &:hover {
+        background-color: black;
+        color: white;
+        border-color: black;
+    }
+
+    &:focus {
+        background-color: black;
+        color: white;
+        border-color: transparent;
+    }
+
+    &:active {
+        background-color: black;
+        color: white;
+        border-color: black;
+    }    
+`
+const previewCss = css`
+    opacity: 0.5;
+`
+const noninteractiveCss = css`
+    &:hover, &:focus, &:active {
+        background-color: #F5F5F5;
+        color: black;
+        border-color:  #F5F5F5;
+    }
+`
+
+interface CustomButtonProps extends ButtonProps {
+    secondary?: boolean
+    selected?: boolean
+    noninteractive?: boolean
+    preview?: boolean
+}
+
+export const UnitButton: React.FC<CustomButtonProps> = ({ secondary, selected, preview, noninteractive, children, ...restProps }) => {
+    const OriginalLabel = children ? children.toString() : ''
+    if (!secondary && OriginalLabel.length > 4) {        
+        let ButtonLabel = OriginalLabel
+        if (!isNaN(+ButtonLabel)) {
+            ButtonLabel = `…${ButtonLabel.substring(ButtonLabel.length - 2)}`
+        } else {
+            ButtonLabel = `${ButtonLabel.substring(0, 2)}…`
+        }
+        return (
+            <Tooltip placement='topLeft' title={OriginalLabel}>
+                <Button css={css`
+                    ${buttonCss};
+                    ${selected ? selectedCss : ''};
+                    ${selected ? selectedCss : ''};
```

Comment:
Duplicates previous string

Reply:
can we use classNames for css classNames composition

Reply:
Skipped "can we use classNames for css classNames composition?" will return to it later
duplicates  - fixed